File: medit-frontend/api-types/Medication.ts
import { MedicationTypeEnum } from "./MedicationTypeEnum";

export interface Medication {
  id?: number;
  userId: number;
  name: string;
  type: MedicationTypeEnum;
  dose: string;
  program: string;
  quantity: number;
  note: string;
  created_at: Date;
  update_at: Date;
  synced_at: Date;
}

File: medit-backend/src/main/kotlin/medicare/medit/model/Medication.kt
package medicare.medit.model

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "medications")
data class Medication(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: User,
    val name: String,
    val type: MedicationTypeEnum,
    val dose: String,
    val program: String,
    val quantity: Int,
    val note: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    var syncedAt: LocalDateTime
)
